app/components/hook-switch.js

```diff
@@ -13,12 +18,12 @@ export default Ember.Component.extend({
       let pusher = this.get('pusher'),
         repoId = hook.get('id');
 
-      return hook.toggle().then(() => {
+      yield hook.toggle().then(() => {
         pusher.subscribe(`repo-${repoId}`);
       }, () => {
         this.toggleProperty('hook.active');
         return this.sendAction('onToggleError', hook);
       });
     }
-  }
+  }),
```

Comment:
Excellent! Does there need to be anything [like `drop` or `restartable`](http://ember-concurrency.com/#/docs/task-concurrency) here?